pymatgen/core/structure.py

```diff
@@ -60,6 +59,7 @@
     from ase.optimize.optimize import Optimizer
     from matgl.ext.ase import TrajectoryObserver
     from numpy.typing import ArrayLike, NDArray
+    from typing_extensions import Self
```

Comment:
So we can only type check on Python 3.10? Is there no ruff rule that can be turned on that does

```suggestion
    try:
        from typing_extensions import Self
    except ImportError:
        from typing import Self
```

or something similar?